src/app/components/layout/footer/footer.component.ts

```diff
@@ -11,10 +9,19 @@ import * as moment from 'moment';
   styleUrls: ['./footer.component.css']
 })
 
-export class FooterComponent {
-  public appVersion: string = environment.VERSION;
+export class FooterComponent implements OnInit {
+  public appVersion;
   public git = data['default'];
   copyrightYear = moment().year();
 
-  constructor(public router: Router) {}
+  constructor(public router: Router) { }
+
+  ngOnInit(): void {
+    if (this.git?.distance == 0) {
+      // released version
+      this.appVersion = `${this.git.version}-${this.git.hash}`
+    } else {
+      this.appVersion = this.git.raw.substr(0, this.git.raw.lastIndexOf('-'));
```

Comment:
can not we simply do,
 this.appVersion = this.git?.distance == 0 ? this.git.raw.replace( '-0-', '-') :  this.git.raw;